### DISCUSSION TITLE
Sitemaps

### DISCUSSION THREAD

**Discussion by FredKSchott** [withastro/roadmap#146] [Proposal]
This topic has come up in a few places, so I wanted to get us all on the same page around sitemaps. Here's how I'm thinking about them heading into v1.0, but I'd like to hear other thoughts, especially from @natemoo-re and @davidrot.

#### Fully Automatic, Zero-Effort Sitemaps:
- **Today:** `sitemap: true` in the config
- **v1.0:** `@astrojs/sitemap` integration
- Integration can support filters, etc, but the idea is that this is low-effort and doesn't require you to set up a `src/pages/sitemap.xml.js` page yourself.

#### Flexible, Dynamic Sitemaps:
- **Today:** Create a `src/pages/sitemap.xml.js` 
- **v1.0:** Create a `src/pages/sitemap.xml.js` (no change, possible today)
- Note: I'm specifically proposing no new features here: if you want to create your own sitemap, then you're on your own to do so using a package like https://www.npmjs.com/package/sitemap, `import.meta.glob`, etc.

I'm not necessarily against the idea of passing some route manifest to dynamic routes like `src/pages/sitemap.xml.js`, but I'm not convinced that the cost of doing this is worth it before v1.0. Specifically, the cost of implementing this before we have SSR since that is going to be a huge change to any concepts that require the idea of "known pages", which would no longer exist in SSR.

**Comment by natemoo-re** [comment]
Thanks for spinning up a discussion!

I'm absolutely on board with this direction. I like the idea of having a simple, straight-forward solution baked in and integrations seem like a great way to improve this API. 

I also like having a `src/pages/sitemap.xml.js` file support for more advanced use cases. I naively expected this to _replace_ our `@astrojs/sitemap` story, but I see how both are valuable. I am very curious how easy it would be to expose a custom `virtual:astro/manifest` import, but I agree that SSR makes a lot of this a bigger question mark. Happy to punt this until post-v1.0.

**Reply by FredKSchott** [reply]
cool!

**Comment by johneatmon** [comment]
Would love to add a stylesheet feature to the sitemap integration, similar to the [styling features available for the RSS feed](https://docs.astro.build/en/guides/rss/#styling). Being able to pass in a custom XSL stylesheet with maybe a sensible default would be tremendous!

**Reply by yavko** [reply]
This is what I was looking for, would love this!

**Reply by sarajw** [reply]
Yes please :)
